src/main/java/peoplesoft/model/job/Rate.java

```diff
@@ -17,14 +20,26 @@
     /**
      * Constructs a {@code Rate} instance.
      *
-     * @param value A value as a double.
+     * @param amount Money per unit time
+     * @param duration Unit time duration
      */
     public Rate(Money amount, Duration duration) {
         requireAllNonNull(amount, duration);
+        requireNonNegative(amount);
+        requirePositive(duration);
         this.amount = amount;
         this.duration = duration;
     }
 
+    /**
+     * Throws NonPositiveDurationException if {@code duration} is not positive.
+     */
+    private void requirePositive(Duration duration) {
+        if (duration.compareTo(Duration.ofHours(0)) <= 0) {
```

Comment:
[`isNegative()`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#isNegative--)

Reply:
oops, you might want to either do `isNegative() || isZero()` or compare against `Duration.ZERO`